src/config.ts

```diff
@@ -1,10 +1,11 @@
 const hasDOM = typeof document !== 'undefined' && typeof window !== 'undefined';
-const scriptElement: HTMLScriptElement = hasDOM && document.currentScript;
+const scriptElement: HTMLOrSVGScriptElement | null = document?.currentScript;
 // @ts-ignore
-const globalConfig: Record<string, PrismConfig[keyof PrismConfig]> =
-	globalThis.Prism?.constructor?.name === 'Object' ? globalThis.Prism : {};
+const globalConfig: Record<string, PrismConfig[keyof PrismConfig] | null> =
+	globalThis.Prism?.constructor?.name === 'Object' ? globalThis.Prism.config : {};
```

Comment:
These are not equivalent. `Prism` here is not the global instance, this is before initialization, so it's just an object literal that folks can define (hence the constructor check)

Reply:
```suggestion
	globalThis.Prism?.constructor?.name === 'Object' ? globalThis.Prism: {};
```